docs/sql-ref-functions-udf-scalar.md

```diff
@@ -1,22 +1,125 @@
 ---
 layout: global
-title: User defined Scalar Functions (UDF)
-displayTitle: User defined Scalar Functions (UDF)
+title: Scalar User Defined Functions (UDF)
+displayTitle: Scalar User Defined Functions (UDF)
 license: |
   Licensed to the Apache Software Foundation (ASF) under one or more
   contributor license agreements.  See the NOTICE file distributed with
   this work for additional information regarding copyright ownership.
   The ASF licenses this file to You under the Apache License, Version 2.0
   (the "License"); you may not use this file except in compliance with
   the License.  You may obtain a copy of the License at
- 
+
      http://www.apache.org/licenses/LICENSE-2.0
- 
+
   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
 ---
 
-**This page is under construction**
+### Description
+
+This documentation contains examples that demonstrate how to define and register UDFs that act on a single row and invoke them in SPARK SQL.
+
+### Examples
+
+{% highlight sql %}
+
+// Define, register a zero argument non-deterministic UDF
+// Scala
+import functions.udf
+val foo = udf(() => Math.random())
+spark.udf.register("random", foo.asNondeterministic())
```

Comment:
How about documenting all the `asXXX` methods somewhere?

Reply:
Do you mean add a comment  to explain asNondeterministic?

Reply:
I didn't mean so, but the comment looks nice.
My original suggestion wasI that, how about adding a subsection (just like `[Parameter](https://github.com/apache/spark/blob/master/docs/sql-ref-syntax-qry-select.md#parameters)`) about the methods `asXXXX`?

Reply:
I see. Since we need a page to document the UDF APIs, I will have a subsection for asXXX.